docs/extensions/overview.md

```diff
@@ -2,109 +2,124 @@
+DuckDB has three types of extensions.
+
+#### Built-In Extensions
+
+Built-in extensions are loaded at startup and are immediately available for use.
+
+```sql
+SELECT * FROM 'test.json';
 ```
 
-The list of supported platforms may increase over time, but the current list of platforms includes:
-* linux_amd64_gcc4
-* linux_amd64
-* linux_arm64
-* osx_amd64
-* osx_arm64
-* windows_amd64
-* windows_amd64_rtools
+This will use the `json` extension to read the JSON file.
 
-See above for a list of extension names and how to pull the latest list of extensions.
+> To make the DuckDB distribution lightweight, it only contains a few fundamental built-in extensions (e.g., `autocomplete`, `json`, `parquet`), which are loaded automatically upon startup.
 
+#### Autoloadable Extensions
 
-## Loading an Extension from Local Storage
+Autoloadable extensions are loaded on first use.
 
-Extensions are stored in gzip format, so they must be unzipped prior to use. 
-There are many methods to decompress gzip. Here is a Python example:
+```sql
+SELECT * FROM 'https://raw.githubusercontent.com/duckdb/duckdb-web/main/data/weather.csv';
+```
+
+To access files via the HTTPS protocol, DuckDB will automatically load the `httpfs` extension.
+Similarly, other autoloadable extensions (`aws`, `fts`) will be loaded on-demand.
+If an extension is not already available locally, it will be installed from the official extension repository (`extensions.duckdb.org`).
 
-```python
-import gzip
-import shutil
+#### Explicitly Loadable Extensions
 
-with gzip.open('httpfs.duckdb_extension.gz','rb') as f_in:
-   with open('httpfs.duckdb_extension', 'wb') as f_out:
-     shutil.copyfileobj(f_in, f_out)
+Some extensions make several changes to the running DuckDB instance, hence, autoloading them may not be possible.
+These extensions have to be installed and loaded using the following SQL statements:
+
+```sql
+INSTALL 'spatial';
+LOAD 'spatial';
 ```
 
-After unzipping, the install and load commands can be used with the path to the .duckdb_extension file. 
-For example, if the file was unzipped into the same directory as where DuckDB is being executed:
 ```sql
-INSTALL 'httpfs.duckdb_extension';
-LOAD 'httpfs.duckdb_extension';
+CREATE TABLE tbl(geom GEOMETRY);
 ```
 
-## Extension Template
+If you are using the [Python API client](../api/python/overview), you can install and load them with the `load_extension(name: str)` and `install_extension(name: str)` methods.
+
+> Autoloadable extensions can also be installed explicitly.
 
-A template for creating extensions is available in the [`extension-template` repository](https://github.com/duckdb/extension-template/). Note that this project is work-in-progress.
+### Ensuring the Integrity of Extensions
 
-## Pages in This Section
+Extensions are signed with a cryptographic key. By default, DuckDB uses its built-in public keys to verify the integrity of extension before loading them.
```

Comment:
```suggestion
Extensions are signed with a cryptographic key, which also simplifies distribution (this is why they are served over `http`, not `https`). By default, DuckDB uses its built-in public keys to verify the integrity of extension before loading them.
```

Reply:
I don't agree with this change

Reply:
Actually, it's fine either way. I will make a paragraph expanding on this in "Working with extensions", but can go in a separate PR.